vehicle/src/Vehicle/Compile/Type.hs

```diff
@@ -189,8 +189,14 @@ typeCheckRecordDef p ident _anns uncheckedType uncheckedFields isUnused = do
     logCompilerSection2 MidDetail pass $
       traverse (checkRecordFieldDef ident) uncheckedFields
 
+  _ <-
+    if isAnnotatedAsTensor anns
```

Comment:
This can now be simplified by using `when` (see other places in the compiler for how it works)